### PR TITLE
pacific: dev/developer_guide: add jaegertracing intial developer documentation

### PR DESCRIPTION
**Pull request by ideepika**
…tion

Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>
(cherry picked from commit 6fd01656104215ed18876eb5635ccbfc54d1de29)

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/dev/developer_guide/jaegertracing.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+isn't recommended for production, but for testing purposes. Let's look at all the
+steps needed:
+
+  1. Update system with Jaeger dependencies, using install-deps::
+
+     $ WITH_JAEGER=true ./install-deps.sh
+
+  2. Compile Ceph with Jaeger enabled:
+
+    - for precompiled build::
```
Comment (zdover23) [inline doc/dev/developer_guide/jaegertracing.rst:49]: I think that these unordered list items need to be indented one more space. They're making the list look crazy:

https://ceph--40343.org.readthedocs.build/en/40343/dev/developer_guide/jaegertracing/

**Review by zdover23 — CHANGES_REQUESTED** [review]
The ordered list needs to have its indentations fixed so that the list builds cleanly.

When that's done, I'll approve this and merge it.

Thanks!

**Review by tchaikov — CHANGES_REQUESTED** [review]
I don't agree on backporting dev docs to stable branches.

**Comment by zdover23** [comment]
> I don't agree on backporting dev docs to stable branches.

I submit to your greater knowledge in this matter. Should we kill this PR?

File: doc/dev/developer_guide/jaegertracing.rst (hunk 1 above)
Comment (ideepika) [reply doc/dev/developer_guide/jaegertracing.rst:49]: you are right! I will fix that in master

**Comment by ideepika** [comment]
> I don't agree on backporting dev docs to stable branches.

since this doc has only been added to master, people who are getting started should not be confused with outdated stable branch docs, which was my primary reason to create this PR, I agree, this should not cost us if we don't include in stable branch.
will reopen if found apt.

**Pull request closed without merging**